exp/api/remote/remote_api.go

```diff
@@ -455,15 +457,21 @@ func SnappyDecompressorMiddleware(logger *slog.Logger) func(http.Handler) http.H
 
 // NewHandler returns HTTP handler that receives Remote Write 2.0
 // protocol https://prometheus.io/docs/specs/remote_write_spec_2_0/.
-func NewHandler(store writeStorage, opts ...HandlerOption) http.Handler {
+func NewHandler(store writeStorage, acceptedMessageTypes MessageTypes, opts ...HandlerOption) http.Handler {
 	o := handlerOpts{
 		logger:      slog.New(nopSlogHandler{}),
 		middlewares: []func(http.Handler) http.Handler{SnappyDecompressorMiddleware(slog.New(nopSlogHandler{}))},
 	}
 	for _, opt := range opts {
 		opt(&o)
 	}
-	h := &handler{opts: o, store: store}
+
+	h := &handler{
+		opts:                 o,
+		store:                store,
+		acceptedMessageTypes: acceptedMessageTypes,
+		acceptedMap:          acceptedMessageTypes.Map(),
```

Comment:
I would just use slice and do the a quick loop lookup, but not a blocker.

Reply:
Ah I kinda copied this from prometheus handler. Should be fine to do loop

Reply:
Ah I kinda copied this from prometheus handler. Should be fine to do loop